[BPK-4246]: BpkTouchableOverlay update (#657)
Skyscanner/backpack-react-native: UNRELEASED.md

> Place your changes below this line.

**Fixed:**

>   - `BpkTouchableOverlay` - fixed radius value to align with new `sm` radius from `xs`

Reviewer: This isn't very readable to a consumer

```suggestion
  - `BpkTouchableOverlay` - updated the `sm` value for `borderRadius` to be `8` (previously `4`).
```
